src/Main.cc

```diff
@@ -45,6 +45,11 @@
 #include <cstdlib>
 #include <iostream>
 #include <string>
+#include <unordered_set>
+
+#define TCL_TRACE_LEAVE_EXEC 2
+#define TCL_TRACE_ENTER_DURING_EXEC 4
```

Comment:
Instead of using a macro to store a constant, use a const variable.
(https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros)